### PR TITLE
Add option to delete clickhouse data

### PR DESCRIPTION
**Pull request by sammacbeth**

### CODE REVIEW

File: reporters/BaseReporter.js [hunk 1]
```diff
@@ -16,7 +16,7 @@ class BaseReporter {
     /**
      * Called whenever site was crawled (either successfully or not)
      * 
-     * @param {{site: string, failures: number, successes: number, urls: number, data: import("../crawler").CollectResult, crawlTimes: Array<Array<number>>, fatalError: Error, numberOfCrawlers: number, regionCode: string}} data 
+     * @param {{site: string, failures: number, successes: number, urls: number, data: import("../crawler").CollectResult | undefined, crawlTimes: Array<Array<number>>, fatalError: Error, numberOfCrawlers: number, regionCode: string}} data
```
Comment (kdzwinel) [inline reporters/BaseReporter.js:19]: Is this related to the change?

File: reporters/ClickhouseReporter.js [hunk 2]
```diff
@@ -100,8 +100,12 @@ const TABLE_DEFINITIONS = [
  */
 function santizeCallArgs(args) {
     // in some cases call args have been stringified, so unwrap that first.
-    const argsArray = typeof args === 'string' ? JSON.parse(args) : (args || []);
-    return argsArray.map((/** @type {string} */ s) => s.replace(/'/g, ''));
+    try {
```
Comment (kdzwinel) [inline reporters/ClickhouseReporter.js:103]: ditto . When this can happen?

File: reporters/ClickhouseReporter.js [hunk 3]
```diff
@@ -148,13 +152,24 @@ class ClickhouseReporter extends BaseReporter {
...
+            .map(table => this.client.query(`ALTER TABLE ${DB}.${table} DELETE WHERE crawlId = '${this.crawlId}'`).toPromise());
+        await Promise.all(deletes);
+        await this.client.query(`ALTER TABLE ${DB}.crawls DELETE WHERE crawlId = '${this.crawlId}'`).toPromise();
+    }
+
     /**
      * Called whenever site was crawled (either successfully or not)
      * 
-     * @param {{site: string, failures: number, successes: number, urls: number, data: import('../crawler').CollectResult, crawlTimes: Array<Array<number>>, fatalError: Error, numberOfCrawlers: number, regionCode: string}} data 
+     * @param {{site: string, failures: number, successes: number, urls: number, data: import('../crawler').CollectResult | undefined, crawlTimes: Array<Array<number>>, fatalError: Error, numberOfCrawlers: number, regionCode: string}} data
```
Comment (kdzwinel) [inline reporters/ClickhouseReporter.js:167]: ditto

**Review by kdzwinel — COMMENTED** [review]
Code LGTM, but there seem to be some unrelated changes that I don't understand.
FWIW I'm only reviewing the code, not actually testing it (since I'm not sure how and since this code is not on a critical path for anything).

File: reporters/BaseReporter.js (hunk 1 above)
Comment (kdzwinel) [reply reporters/BaseReporter.js:19]: Ah, I see. Shouldn't TS complain about that? It seems like it could have figure it out from the code, no?

Comment (sammacbeth) [reply reporters/BaseReporter.js:19]: I'm not sure how strict the TS checking is when the types are in jsdoc. 🤷

File: reporters/ClickhouseReporter.js (hunk 2 above)
Comment (sammacbeth) [reply reporters/ClickhouseReporter.js]: I removed this, as I couldn't repro the issue that triggered the error.

**Review by kdzwinel — APPROVED** [review]
LGTM! Thanks for addressing my feedback.
